File: atlas-extract-server/Dtos/ExtractionJsonRes.cs
namespace App.Dtos;



public class ExtractionBatchUrlRes
{
    public int BatchIndex { get; set; }
    public int StartPage { get; set; }
    public int EndPage { get; set; }
    public string Url { get; set; } = "";
}

public class ExtractionJsonRes
{
    public string ExtractionId { get; set; } = "";
    public List<ExtractionBatchUrlRes> Batches { get; set; } = [];
}

File: atlas-extract-client/src/app/types/ExtractionJsonRes.ts
import { ExtractedItem } from './ExtractedItem';

export interface ExtractionBatchUrl {
    batchIndex: number;
    startPage: number;
    endPage: number;
    url: string;
}

export interface ExtractionJsonRes {
    extractionId: string;
    batches: ExtractionBatchUrl[];
}

export interface ExtractionBatchResult {
    batchIndex: number;
    startPage: number;
    endPage: number;
    s3ResultKey: string;
    items: ExtractedItem[];
}
